play_media/play_media.cpp

```diff
@@ -0,0 +1,243 @@
+// play_media.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include "play_media.h"
+#include <CommCtrl.h >
+
+#define ID_PLAY_MEDIA_BTN 11
+#define ID_PLAY_MEDIA_SLIDER 12
+
+void PlayMedia::updateChannelPublishOptions(bool publishMediaPlayer)
+{
+	// You use ChannelMediaOptions to change channel media options.
+	ChannelMediaOptions channelOptions;
+	channelOptions.publishMediaPlayerAudioTrack = publishMediaPlayer;
+	channelOptions.publishMediaPlayerVideoTrack = publishMediaPlayer;
+	channelOptions.publishMicrophoneTrack = !publishMediaPlayer;
+	channelOptions.publishCameraTrack = !publishMediaPlayer;
+	if (publishMediaPlayer) channelOptions.publishMediaPlayerId = mediaPlayer->getMediaPlayerId();
+	agoraEngine->updateChannelMediaOptions(channelOptions);
+}
+
+void PlayMedia::setupLocalVideo(bool forMediaPlayer)
+{
+	// Pass the window handle to the engine so that it renders the local video.
+	if (forMediaPlayer)
+	{
+		// Stop previewing the local video.
+		agoraEngine->stopPreview();
+		// To display the media file output, pass the local view window handle to the setView method of the IMediaPlayer class.
+		mediaPlayer->setView(gui->localView);
+	}
+	else
+	{
+		// Start the local video preview.
+		agoraEngine->startPreview();
+	}
+}
+
+void PlayMedia::createSpecificGui(HWND& guiWindowReference)
+{
+    HWND parentWindow = guiWindowReference;
+    RECT rect;
+    GetClientRect(parentWindow, &rect);
+    int width = rect.right - rect.left;
+    int height = rect.bottom - rect.top;
+
+    // Calculate general button dimensions
+    int btnWidth = 100;
+    int btnHeight = 30;
+    int btnMargin = 10;
+
+    // Echo Test Button
```

Comment:
Corrected.

Reply:
"Is it not possible to do this in CPP ? "  --- I am not sure  and checking/working  to change  it in parallel. But  at the same time  continue the current structure to progress my item  as  the initial projects 'get_started' , call_policy,etc are approved in the same way. As I made the way as you suggested , it would be easy for me to move all in new way. Thanks.